### PR TITLE
KT-13558 Add quickfix to 'create companion object and member\extension function'

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-13558

### CODE REVIEW

**Comment by klunnii** [comment]
When a receiver class has no companion object it's generated with needless parenthesis. Would it be possible to fix it?
Other than that I noticed that formatting in IDE and test environment differs. IDE one looks as follows:
```
class B {
    companion object {
} // <= moved left
}
```

**Comment by klunnii** [comment]
Please, take a look at the following tests (failed):
- org.jetbrains.kotlin.idea.stubs.MultiFileHighlightingTestGenerated#testEnumReference
- org.jetbrains.kotlin.idea.quickfix.QuickFixTestGenerated.CreateFromUsage.CreateClass.ReferenceExpression#testObjectForEnumInEquals
- org.jetbrains.kotlin.idea.quickfix.QuickFixTestGenerated.CreateFromUsage.CreateClass.ReferenceExpression#testObjectForEnumInArgument

**Comment by t-kameyama** [comment]
Fixed, please take another look.

> Please, take a look at the following tests (failed):
> 
> * org.jetbrains.kotlin.idea.stubs.MultiFileHighlightingTestGenerated#testEnumReference
> * org.jetbrains.kotlin.idea.quickfix.QuickFixTestGenerated.CreateFromUsage.CreateClass.ReferenceExpression#testObjectForEnumInEquals
> * org.jetbrains.kotlin.idea.quickfix.QuickFixTestGenerated.CreateFromUsage.CreateClass.ReferenceExpression#testObjectForEnumInArgument

d292405

> When a receiver class has no companion object it's generated with needless parenthesis. Would it be possible to fix it?
> Other than that I noticed that formatting in IDE and test environment differs. IDE one looks as follows:
> 
> ```
> class B {
>     companion object {
> } // <= moved left
> }
> ```

5c142f9

> I would suggest adding the following tests:
> 
> * receiver class resides in another file

Already exists:
[funOnClassObject.kt](https://github.com/t-kameyama/kotlin/blob/a1e86e8bfaa933b72331fc71b6605e3c1476771e/idea/testData/quickfix/createFromUsage/createFunction/call/funOnClassObject.kt)
[valOnClassObject.kt](https://github.com/t-kameyama/kotlin/blob/a1e86e8bfaa933b72331fc71b6605e3c1476771e/idea/testData/quickfix/createFromUsage/createVariable/property/valOnClassObject.kt)


> * generating property/fun for the existing companion

d8ff30c

File: idea/testData/quickfix/createFromUsage/createVariable/property/valOnClassNoClassObject3.after.1.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class A {
+    companion object {
+        val foo: Int
```
Comment (klunnii) [inline idea/testData/quickfix/createFromUsage/createVariable/property/valOnClassNoClassObject3.after.1.kt:3]: Property has no initializer

Comment (t-kameyama) [reply idea/testData/quickfix/createFromUsage/createVariable/property/valOnClassNoClassObject3.after.1.kt:3]: Other existing test cases also have no initializer. For example:
[privateForMembers.kt.after](https://github.com/t-kameyama/kotlin/blob/837e46a9b88e20044153cfd959d5ff2c1af74154/idea/testData/quickfix/createFromUsage/createVariable/property/privateForMembers.kt.after#L4)
[thisInClass.kt.after](https://github.com/t-kameyama/kotlin/blob/a1e86e8bfaa933b72331fc71b6605e3c1476771e/idea/testData/quickfix/createFromUsage/createVariable/property/thisInClass.kt.after#L5)

Should I add initializer in other existing test cases as well?

**Comment by t-kameyama** [comment]
https://github.com/JetBrains/intellij-community/pull/1613

**Pull request closed without merging**